spec/controllers/users_controller_spec.rb

```diff
@@ -2,8 +2,46 @@
 
 describe UsersController, type: :controller do
 
-  let!(:admin) {create(:user, :admin)}
-  let!(:hacker) {create(:user)}
+  let!(:admin) {create(:user, :admin, name: 'Admin')}
+  let!(:hacker) {create(:user, name: 'Hacker')}
+
+  describe 'GET index' do
+
+    context 'default index list' do
+      before :each do
+        allow(controller).to receive(:current_user) { admin }
+      end
+
+      it "returns all users" do
+        smith = create(:user, name: 'Jane Smith')
+        jones = create(:user, name: 'Frank Jones')
+        get :index
+        expect(assigns(:users)).to match_array([jones, smith, admin, hacker])
+      end
+    end
+
+    context 'name search' do
+      before :each do
+        allow(controller).to receive(:current_user) { admin }
+      end
+      it "returns the searched for user" do
+        smith = create(:user, name: 'Jane Smithbot')
+        jones = create(:user, name: 'Frank Jones')
+        get :index , search: 'Smithbot'
+        expect(assigns(:users)).to match_array([smith])
+      end
+    end
+
+    context 'location search' do
+      before :each do
+        allow(controller).to receive(:current_user) { admin }
+      end
+      it "returns users near the searched location" do
+        get :index, location: '21224'
+        expect(User).to receive(:near).and_call_original
```

Comment:
swap these 2 lines, you have to say "expect this thing to receive" BEFORE you actually make the call  :tongue: